src/components/NotificationModal.tsx

```diff
@@ -0,0 +1,200 @@
+    subscriptionMethodControls,
+    triggers,
+    resourceType,
+    resourceId,
+    onCancel,
+  });
+
+  const currentMethodControl = subscriptionMethodControls[
+    selectedSubscriptionMethod
+  ] as SubscriptionMethodControl;
+  const label = get(currentMethodControl, "label");
+  const placeholder = get(currentMethodControl, "placeholder");
+  const targetPath = get(currentMethodControl, "targetPath");
+
+  return (
+    <Modal
+      data-test-id="subscription-modal"
+      title={<H2>Add Subscription</H2>}
+      visible={visible}
+      onCancel={onCancel}
+      footer={[
+        <Button
+          key="cancel"
+          onClick={onCancel}
+          dataCy="cancel-subscription-button"
+        >
+          Cancel
+        </Button>,
+        <Button
+          key="save"
+          dataCy="save-subscription-button"
+          loading={mutationLoading}
+          disabled={!isFormValid}
+          onClick={onClickSave}
+          variant="danger"
+        >
+          Save
+        </Button>,
+      ]}
+      width="50%"
+      wrapProps={{
+        "data-cy": "task-notification-modal",
+      }}
+    >
+      <>
+        when
+        <StyledSelect
+          value={selectedTriggerId}
+          onChange={(v: string) => {
+            setSelectedTriggerId(v);
+          }}
+          data-test-id="when-select"
+        >
+          {triggers.map((t) => (
+            <Option
+              key={t.trigger}
+              value={t.trigger}
+              data-test-id={`${t.trigger}-option`}
+            >
+              {t.label}
+            </Option>
+          ))}
+        </StyledSelect>
+      </>
+      {(extraFields ?? []).map(({ text, key }) => (
+        <ExtraFieldContainer key={key}>
+          <span>{text}</span>
+          <StyledInputNumber
+            data-test-id={`${key}-input-number`}
+            type="number"
+            value={extraFieldInputVals[key]}
+            onChange={(n) => {
+              setExtraFieldInputVals({
+                ...extraFieldInputVals,
+                [key]: n,
+              });
+            }}
+          />
+        </ExtraFieldContainer>
+      ))}
+      <div>
+        then notify by:
+        <StyledSelect
+          data-test-id="notify-by-select"
+          value={selectedSubscriptionMethod}
+          onChange={(v: string) => {
+            setSelectedSubscriptionMethod(v);
+          }}
+        >
+          {subscriptionMethods.map((s) => (
+            <Option
+              key={s.value}
+              value={s.value}
+              data-test-id={`${s.value}-option`}
+            >
+              {s.label}
+            </Option>
+          ))}
+        </StyledSelect>
+      </div>
+      <div>
+        {currentMethodControl && (
+          <>
+            {label}
+            <StyledInput
+              placeholder={placeholder}
+              data-test-id={`${targetPath}-input`}
+              onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
If set, the email/slack username for the user should default in here (feel free to defer this to later)

Reply:
https://jira.mongodb.org/browse/EVG-8070